presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -132,51 +139,95 @@ public void testOffsetPushDown() throws ExecutionException, InterruptedException
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(2);
     }
 
-    @Test
+    @Test(retryAnalyzer = RetryAnalyzerForTimestampTest.class)
     public void testTimestampLogAppendModePushDown() throws ExecutionException, InterruptedException
     {
-        Pair<String, String> timePair = createTimestampTestMessages(topicNameLogAppend);
+        RecordMessage recordMessage = createTimestampTestMessages(topicNameLogAppend);
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
         String sql = String.format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameLogAppend, timePair.first(), timePair.second());
+                topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime());
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameLogAppend, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(2);
+    }
+
+    private String dumpResultAndData(String topic, RecordMessage recordMessage, DistributedQueryRunner queryRunner, String sql, ResultWithQueryId<MaterializedResult> queryResult)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("dump 5 expected messages:").append("\n");
+        for (int i = 0; i < recordMessage.getDumpMessages().size(); i++) {
+            String dumpMessage = recordMessage.getDumpMessages().get(i);
+            sb.append("  ").append(dumpMessage);
+            if (i == TIMESTAMP_TEST_START_INDEX) {
+                sb.append(", startTime:").append(recordMessage.getStartTime());
+            }
+            else if (i == TIMESTAMP_TEST_END_INDEX) {
+                sb.append(", endTime:").append(recordMessage.getEndTime());
+            }
+            sb.append("\n");
+        }
+        sb.append(String.format("dump test sql:%s\n", sql));
+        sb.append("dump test sql result:").append("\n").append(dumpResult(queryResult.getResult()));
+        sql = String.format("SELECT _partition_id,_partition_offset,_timestamp FROM default.%s WHERE _partition_offset between %s and %s order by _partition_id, _timestamp",
```

Comment:
If you are breaking parameter list then put each parameter in separate line please.